jib-core/src/test/java/com/google/cloud/tools/jib/docker/DockerClientTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.docker;
+
+import com.google.cloud.tools.jib.blob.Blobs;
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link DockerClient}. */
+@RunWith(MockitoJUnitRunner.class)
+public class DockerClientTest {
```

Comment:
Should we add a test for `isDockerInstalled()`? (Is it even possible to test it?)

Reply:
Not too sure how to test this. Any good ideas?

Reply:
I guess the main thing we'd want to test is making sure running a command throws an IO exception if the command is missing, so maybe we could use mocks to start a `ProcessBuilder` with a local 
"fake docker" script in one test, and ensure it fails when we try to start it with a non-existent script?

Reply:
Not sure if `ProcessBuilder` can run local scripts or how this would pan out as far as cross-platform test support goes, etc.